test/connectors/HeaderGeneratorSpec.scala

```diff
@@ -22,39 +22,44 @@ import org.mockito.Mockito.{mock, when}
 import org.scalatest.wordspec.AnyWordSpec
 import play.api.http.HeaderNames
 import uk.gov.hmrc.http.HeaderCarrier
+import utils.UUIDGenerator
 
 
 class HeaderGeneratorSpec extends AnyWordSpec with HeaderNames with SpecBase {
 
   val mockAppConfig: AppConfig = mock(classOf[AppConfig])
-  val testHeaderGenerator = new HeaderGenerator(mockAppConfig)
+  val mockUUIDGenerator: UUIDGenerator = mock(classOf[UUIDGenerator])
+  val testHeaderGenerator = new HeaderGenerator(mockAppConfig,mockUUIDGenerator)
   val bearerToken = "someToken"
+  val correlationId = "someUUID"
   val env = "someEnv"
 
   "HeaderGenerator.headersForPEGA" must {
 
     "create headers" in {
-      when(mockAppConfig.pegaEnvironment) thenReturn env
```

Comment:
Is this no longer needed? If so will need to remove "headers("Environment") shouldBe env" on line 44